packages/web/src/pages/sign-up-page/utils/handleSchema.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  HandleCheckStatus,
+  MAX_HANDLE_LENGTH,
+  signUpFetch
+} from '@audius/common'
+import { z } from 'zod'
+
+import { audiusQueryContext } from 'services/audiusQueryContext'
+import restrictedHandles from 'utils/restrictedHandles'
+
+export const errorMessages = {
+  badCharacterError: 'Please only use A-Z, 0-9, . and _',
+  twitterReservedError: 'This verified Twitter handle is reserved.',
+  instagramReservedError: 'This verified Instagram handle is reserved.',
+  tiktokReservedError: 'This verified TikTok handle is reserved.',
+  genericReservedError: 'This verified handle is reserved.',
+  unknownError: 'An unknown error occurred.',
+  handleTakenError: 'That handle has already been taken.',
+  missingHandleError: 'Please enter a handle.'
+}
+
+export const handleSchema = z.object({
+  handle: z
+    .string()
+    .max(MAX_HANDLE_LENGTH)
+    .regex(/^[a-zA-Z0-9_.]*$/, errorMessages.badCharacterError)
+    .refine(
+      (h) => !restrictedHandles.has(h.toLowerCase()),
+      errorMessages.handleTakenError
+    )
+    .superRefine(async (h, context) => {
```

Comment:
Super refine that's chill, is this for async or something?

Reply:
it gives you more control over the error handling, e.g. you can set different messages for different outcomes and you can make it abort the rest of the checks if one fails